bulk_grades/api.py

```diff
@@ -331,6 +332,91 @@ def get_rows_to_export(self):
+        """
+        Create InterventionCSVProcessor.
+        """
+        self.columns = ['user_id', 'username', 'email', 'student_key', 'full_name', 'course_id', 'track', 'cohort',
+                        'number of videos overall', 'number of videos last week', 'number of problems overall',
+                        'number of problems last week',
+                        'number of correct problems overall', 'number of correct problems last week',
+                        'number of problem attempts overall', 'number of problem attempts last week',
+                        'number of forum posts overall', 'number of forum posts last week',
+                        'date last active']
+        self.course_id = None
+        self.track = self.cohort = None
+        super(InterventionCSVProcessor, self).__init__(**kwargs)
+        self._course_key = CourseKey.from_string(self.course_id)
+        self._subsections = self._get_graded_subsections(self._course_key)
+        self.columns.append('course grade letter')
+        self.columns.append('course grade numeric')
+
+    def _get_graded_subsections(self, course_id):
+        """
+        Return list of graded subsections.
+        """
+        subsections = {}
+        for subsection in grades_api.graded_subsections_for_course_id(course_id):
+            short_block_id = subsection.location.block_id[:8]
+            if short_block_id not in subsections:
+                for key in ('name', 'grade'):
+                    self.columns.append('{}-{}'.format(key, short_block_id))
+                subsections[short_block_id] = (subsection, subsection.display_name)
+        return subsections
+
+    def get_rows_to_export(self):
+        """
+        Return iterator of rows to export.
+        """
+        enrollments = list(_get_enrollments(self._course_key, track=self.track, cohort=self.cohort))
+        grades_api.prefetch_course_and_subsection_grades(self._course_key, [enroll['user'] for enroll in enrollments])
+        client = LearnerAPIClient()
+        intervention_data = client.courses(self.course_id).intervention().get()
+        for enrollment in enrollments:
+            cohort = get_cohort(enrollment['user'], self._course_key, assign=False)
+            course_grade = grades_api.CourseGradeFactory().read(enrollment['user'], course_key=self._course_key)
+            int_user = intervention_data.get(enrollment['user'].username)
+            row = {
+                'user_id': enrollment['user_id'],
+                'username': enrollment['username'],
+                'email': enrollment['user'].email,
+                'student_key': enrollment['student_uid'],
+                'full_name': enrollment['full_name'],
+                'track': enrollment['track'],
+                'course_id': self.course_id,
+                'cohort': cohort.name if cohort else None,
+                'number of videos overall': int_user['videos_overall'] if int_user is not None else 0,
```

Comment:
still has a lot of repetition. Why not just one branch?

Reply:
Can you explain what you mean by "one branch" ?